internal/streamingnode/server/wal/interceptors/timetick/ack/ack.go

```diff
@@ -0,0 +1,87 @@
+package ack
+
+import (
+	"go.uber.org/atomic"
+
+	"github.com/milvus-io/milvus/pkg/streaming/util/message"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+)
+
+var _ typeutil.HeapInterface = (*timestampWithAckArray)(nil)
+
+// newAcker creates a new acker.
+func newAcker(ts uint64, lastConfirmedMessageID message.MessageID) *Acker {
+	return &Acker{
+		acknowledged: atomic.NewBool(false),
+		detail:       newAckDetail(ts, lastConfirmedMessageID),
+	}
+}
+
+// Acker records the timestamp and last confirmed message id that has not been acknowledged.
+type Acker struct {
+	acknowledged *atomic.Bool // is acknowledged.
+	detail       *AckDetail   // info is available after acknowledged.
+}
+
+// LastConfirmedMessageID returns the last confirmed message id.
+func (ta *Acker) LastConfirmedMessageID() message.MessageID {
+	return ta.detail.LastConfirmedMessageID
+}
+
+// Timestamp returns the timestamp.
+func (ta *Acker) Timestamp() uint64 {
+	return ta.detail.Timestamp
+}
+
+// Ack marks the timestamp as acknowledged.
+func (ta *Acker) Ack(opts ...AckOption) {
+	for _, opt := range opts {
+		opt(ta.detail)
+	}
+	ta.acknowledged.Store(true)
```

Comment:
acknowledged should not be true if  the error is not nil with ack option

Reply:
The acknowledgement here is for timetick manager.
All allocated timetick must be acked, Otherwise, the timetick interceptor based on heap got stuck.
The error here need to be handled by outside AckDetail management, but current behaviour is just ignored, and drop the illegal message at scanner side.